veniq/baselines/semi/alternatives/all_opportunities/create_all_opportunities.py

```diff
@@ -0,0 +1,41 @@
+from typing import Dict, List
+
+from veniq.ast_framework import AST
+from ...extract_semantic import extract_method_statements_semantic
+from ..._common_cli import common_cli
+from ..._common_types import Statement, StatementSemantic, ExtractionOpportunity
+
+
+def create_extraction_opportunities(
+    statements_semantic: Dict[Statement, StatementSemantic]
+) -> List[ExtractionOpportunity]:
+    statements = list(statements_semantic.keys())
+    extraction_opportunities: List[ExtractionOpportunity] = []
+    for first in range(len(statements)):
+        for last in range(first, len(statements)):
+            extraction_opportunities.append(tuple(statements[first: last + 1]))
+    return extraction_opportunities
+
+
+def _print_extraction_opportunities(method_ast: AST, filepath: str, class_name: str, method_name: str):
```

Comment:
- Two different modules have functions with the name `_print_extraction_opportunities` and an exact list of arguments
- These functions have common functionality

Can we refactor it to reduce code duplication?

Reply:
Why do you think that extracting code duplication is **always** improve code quality? Can you imagine how understandable the new function will be?

Reply:
I think that not always. But in this case may be. And I ask you is it possible?  I see at least an opportunity to extract these two lines: 

```
statements_semantic = extract_method_statements_semantic(method_ast)
extraction_opportunities = create_extraction_opportunities(statements_semantic)
```

What do you think?

Reply:
It is possible, but the complexity of the functions after such refactoring will increase and it will be harder to read them. You will have to jump over one definition to another to understand the full process. I think we better leave duplication here, because `main` part of each module always differ a bit from the other one and unifying it all in one function will make the code harder to read.

Reply:
Let's imagine we extract these two lines to separate method and call it. How complexity is going to increase?